src/api/notificationTypes.js

```diff
@@ -3,42 +3,74 @@
 import { type UserId } from './idTypes';
 
 // This format is as of 2020-02, commit 3.0~3347.
-type BaseData = $ReadOnly<{|
-  message_ids: [number], // single-element tuple!
-
-  realm_uri: string, // as in `/server_settings` response
+type BaseData = {|
+  /** The realm URL, same as in the `/server_settings` response. */
+  +realm_uri: string,
 
   // The server and realm_id are an obsolete substitute for realm_uri.
-  server: string, // settings.EXTERNAL_HOST
-  realm_id: number, // server-internal realm identifier
+  -server: string, // settings.EXTERNAL_HOST
+  -realm_id: number, // server-internal realm identifier
```

Comment:
Interesting. The `-` marks these properties as _write_-only. We do that…not because we'd ever want to write to them (we wouldn't), right, but because we want Flow to shout at us if we try to read them (seeing as they're obsolete)?

Reply:
Right. This lets the type still document that they might be there, and it lets them appear when we construct one of these objects, like in tests, but prevents reading them.

We have another example of this on `Subscription`:
```js
  /** (To interpret this value, see `getIsNotificationEnabled`.) */
  +push_notifications: null | boolean,

  // To meaningfully interpret these, we'll need logic similar to that for
  // `push_notifications`.  Pending that, the `-` ensures we don't read them.
  -audible_notifications: boolean,
  -desktop_notifications: boolean,
|};
```

Reply:
> it lets them appear when we construct one of these objects, like in tests

Mm, right, I forgot about tests.

I guess if we have tests that use one of these objects, and we say that the object is of the `ApnsPayload` type, we'll _have_ to include these write-only fields. That's because they're not marked as optional. I guess that's how we say that the server is still sending them…though it seems like servers don't need to be sending them.

Reply:
True. It is kind of nice to have this sort of property in objects we use in test data, because it makes the test data representative of what real servers send... but when it's one that's totally obsolete like this `server` and `realm_id`, it also risks just being noise. So if it comes up, I wouldn't be sad to just comment them out, like we do for `Message#recipient_id`:
```js
  /** Deprecated; a server implementation detail not useful in a client. */
  // recipient_id: number,
```

Commenting it out makes our test data less representative, so that effectively we're trusting that our code doesn't get confused by an unexpected extra property. In general we don't write code in patterns that would get confused by that -- I'm not even sure offhand what a realistic such pattern would be -- so that's acceptable.

OTOH for something like `audible_notifications` and `desktop_notifications`, they totally are something we'll want to handle in the future, and we're just marking them as unusable so one sees the comment before trying to use them. So those have a stronger reason to be present, as `-`, rather than deleted or commented out.